main.py

```diff
@@ -20,107 +21,155 @@ class DnsInfo(TypedDict):
     modified_on: str
 
 
-def list_zones() -> list[DnsInfo]:
-    '''List all zones associated with this cloudflare API key'''
+def list_zones() -> List[DnsInfo]:
+    """list_zones.
 
-    endpoint = 'zones'
+    List all zones associated with this cloudflare API key
+
+    Args:
+
+    Returns:
+        List[DnsInfo]:
+    """
+
+    endpoint = "zones"
     r = requests.get(api_url + endpoint, headers=headers)
 
     if r.status_code == 200:
-        return r.json()['result']
+        return r.json()["result"]
 
     else:
-        logging.error('HTTP: error listing domain')
+        logging.error("HTTP: error listing domain")
         logging.debug(r)
         raise RuntimeError("Listing DNS did not return 200")
 
 
+class NoZoneDataError(Exception):
+    ...
+
+
 def get_zone_id_by_name(name: str) -> str:
-    '''Get a zone id associated with API key and domain name'''
+    """Get a zone id associated with API key and domain name"""
 
     zone_data = list_zones()
     for zone in zone_data:
-        if zone['name'] == name:
-            return zone['id']
+        if zone["name"] == name:
+            return zone["id"]
+    raise NoZoneDataError
 
 
-def dns_info(zone_id: str) -> dict:
-    '''Get all dns information related to this zone id'''
+# NP - What is the structure of the returned dictionaory?
+def dns_info(zone_id: str) -> Dict[Any, Any]:
+    """Get all dns information related to this zone id"""
 
-    endpoint = 'zones/' + zone_id + '/dns_records'
+    endpoint = "zones/" + zone_id + "/dns_records"
     r = requests.get(api_url + endpoint, headers=headers)
 
     if r.status_code == 200:
         return r.json()
 
     else:
-        logging.error('HTTP: error getting dns info')
+        logging.error("HTTP: error getting dns info")
         logging.debug(r)
         raise RuntimeError("Getting DNS info did not return 200")
 
 
-def get_a_record_ips(zone_id: str) -> list[str]:
-    '''Get all ip address associated with all A-records on this zone id'''
+def get_a_record_ips(zone_id: str) -> List[str]:
+    """Get all ip address associated with all A-records on this zone id"""
 
     r = dns_info(zone_id)
     ip_list = []
-    for record in r['result']:
-        if record['type'] == 'A':
-            ip_list.append(record['content'])
+    res: List[DnsInfo] = r["result"]
```

Comment:
type hinting here made LSP a bit happier than without it since dns_info is typed with Dict[Any, Any]